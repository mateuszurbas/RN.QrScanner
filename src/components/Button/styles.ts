import { Colors } from '@/theme/colors';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: 115,
    paddingHorizontal: 12,
  },
  primary: {
    backgroundColor: Colors.blue,
    paddingVertical: 11,
  },
  secondary: {
    backgroundColor: Colors.pink,
    paddingVertical: 9,
  },
  disabled: {
    opacity: 0.6,
  },
  icon: {
    marginRight: 8,
  },
  text: {
    color: Colors.white,
  },
  primaryText: {
    fontSize: 15,
  },
  secondaryText: {
    fontSize: 14,
  },
});

export default styles;
