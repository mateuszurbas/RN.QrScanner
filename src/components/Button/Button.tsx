import React, { memo, ReactNode } from 'react';
import { StyleProp, TouchableOpacity, View, ViewStyle } from 'react-native';

import styles from './styles';
import { AppText } from '../AppText';

export enum ButtonVariants {
  primary = 'primary',
  secondary = 'secondary',
}

type Props = {
  label: string;
  onPress: VoidFunction;
  variant?: ButtonVariants;
  icon?: ReactNode;
  disabled?: boolean;
  style?: StyleProp<ViewStyle> | StyleProp<ViewStyle>[];
  textStyle?: StyleProp<ViewStyle> | StyleProp<ViewStyle>[];
};

const Button = ({
  label,
  onPress,
  variant = ButtonVariants.primary,
  icon,
  disabled,
  style,
  textStyle,
}: Props) => {
  return (
    <TouchableOpacity
      accessibilityRole="button"
      accessibilityLabel={label}
      accessibilityHint={label}
      style={[
        styles.container,
        styles[variant],
        disabled && styles.disabled,
        style,
      ]}
      disabled={disabled}
      onPress={onPress}>
      {icon && <View style={styles.icon}>{icon}</View>}
      <AppText style={[styles.text, styles[`${variant}Text`], textStyle]}>
        {label}
      </AppText>
    </TouchableOpacity>
  );
};

export default memo(Button);
