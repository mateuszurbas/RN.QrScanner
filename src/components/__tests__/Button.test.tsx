import * as React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../Button';

it(`Component Button renders correctly`, () => {
  const tree = renderer.create(<Button label="Test" onPress={() => {}} />);
  expect(tree).toMatchSnapshot();
});
