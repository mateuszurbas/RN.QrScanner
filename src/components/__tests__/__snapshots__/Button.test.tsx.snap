// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button renders correctly 1`] = `
<View
  accessibilityHint="Test"
  accessibilityLabel="Test"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#06B6D4",
      "borderRadius": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 115,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 11,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto_500Medium",
        },
        Array [
          Object {
            "color": "#FFFFFF",
          },
          Object {
            "fontSize": 15,
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;
